app/models/service_template.rb

```diff
@@ -207,36 +213,35 @@ def create_service(service_task, parent_svc = nil)
   end
 
   def set_service_type
```

Comment:
@bzwei The logic of this method has been refactored and it would be good to add some tests around it.

Some additional thoughts:
1. This should be a private method
2. Change logic to return if immediately if the current service_type is `internal` and leave the previous logic in place.  (Assuming that's why these changes were made.)
3. Potentially rename method to avoid using "set" prefix which robocop usually warns about.